modules/dns_virsh/dns_virsh_main.rb

```diff
@@ -7,45 +8,46 @@ class Record < ::Proxy::Dns::Record
     include Proxy::Util
     include Proxy::Virsh
 
-    def self.record(attrs = {})
-      new(attrs.merge(:virsh_network => ::Proxy::SETTINGS.virsh_network))
-    end
-
-
-    def initialize options = {}
-      @network = options[:virsh_network]
+    def initialize(network = nil)
+      @network = network || ::Proxy::SETTINGS.virsh_network
       raise "DNS virsh provider needs 'virsh_network' option" unless network
```

Comment:
Ah, I see what's happening - the local `network` variable is hiding the attr_reader defined by Proxy::Virsh.